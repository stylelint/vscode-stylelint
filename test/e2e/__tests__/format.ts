import * as assert from 'node:assert/strict';

import { commands } from 'vscode';

import { openDocument, closeAllEditors, setupSettings, sleep } from '../helpers';

describe('Document formatting', () => {
	setupSettings({ '[css]': { 'editor.defaultFormatter': 'stylelint.vscode-stylelint' } });

	afterEach(async () => {
		await closeAllEditors();
	});

	// eslint-disable-next-line @typescript-eslint/no-require-imports
	const isStylelint16 = require('stylelint/package.json').version.startsWith('16.');

	// Only Stylelint before 16 had formatting related rules
	const pre16It = isStylelint16 ? it.skip : it;

	pre16It('should format document using formatting options', async () => {
		const editor = await openDocument('defaults/format.css');

		editor.options.tabSize = 4;
		editor.options.insertSpaces = false;

		await sleep(1000); // HACK: Prevent flaky test.
		await commands.executeCommand('editor.action.formatDocument');

		assert.equal(
			editor.document.getText(),
			`a {
	color: #fff;
}
`,
		);
	});
});
