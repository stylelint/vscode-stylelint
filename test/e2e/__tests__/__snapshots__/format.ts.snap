// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document formatting should format document using formatting options 1`] = `
"a {
	color: #fff;
}
"
`;
